src/muse/constraints.py

```diff
@@ -493,6 +494,64 @@ def max_production(
     )
 
 
+@register_constraints
+def demand_limitting_capacity(
+    demand_: xr.DataArray,
+    assets: xr.Dataset,
+    search_space: xr.DataArray,
+    market: xr.Dataset,
+    technologies: xr.Dataset,
+    year: int | None = None,
+) -> Constraint:
+    """Limits the maximum combined capacity to match the demand.
+
+    This is a somewhat more restrictive constraint than the max_production constraint or
+    the maximum capacity expansion. In this case, the combined new capacity of all
+    assets must be sufficient to meet the demand of the most demanding timeslice, and
+    no more.
+
+    Rather than coding from scratch the constraint, we can use the max_production
+    constraint and the demand constraint to construct this constraint. Starting from
+    the maximum production instead of the maximum capacity ensures that the constraint
+    accounts for the utilization factor of the technologies.
+    """
+    # We start with the maximum production constraint and the demand constraint
+    capacity_constraint = max_production(
+        demand_, assets, search_space, market, technologies, year=year
+    )
+    demand_constraint = demand(
+        demand_, assets, search_space, market, technologies, year=year
+    )
+
+    # We need to find the most demanding timeslice and use as the demand constraint
+    b = (
+        demand_constraint.b.max("timeslice")
+        if "timeslice" in demand_constraint.b.dims
+        else demand_constraint.b
+    )
+
+    # Now we need to find the maximum capacity constraint (as the capacity here is
+    # negative), and switch the sign to make it positive. In practice, we want to
```

Comment:
Why is capacity negative here? I realise there's a minus sign in the final line of `max_production`, but I don't really understand why that's done...

Reply:
Again, not capacity, but a multiplier to to the capacity. And it is negative because we are using as the starting point the maximum production to avoid repeat Bing the same calculation. In there, the capacity related factor is negative given that the production related factor is positive, but here we need it positive, so the equation "Cf x Capacity <= Demand" has a solution with a positive capacity.